File: src/022/022.go
package main

import (
	"fmt"
	"io/ioutil"
	"sort"
	"strings"
)

func loadNames() []string {
	file, err := ioutil.ReadFile("p022_names.txt")

	if err != nil {
		fmt.Println("Error reading file.")
	}

	namesString := strings.Replace(string(file), "\"", "", -1)
	names := strings.Split(namesString, ",")
	sort.Strings(names)
	return names
}

func solve() int {
	names := loadNames()
	numberOfNames := len(names)

	sum := 0
	// Iterate over names
	for i := 0; i < numberOfNames; i++ {
		nameScore := 0
		nameSize := len(names[i])

		// Chars of name
		for j := 0; j < nameSize; j++ {
			nameScore += int(names[i][j]) - 64
		}

		sum += nameScore * (i + 1)
	}

	return sum
}

func main() {
	result := solve()
	fmt.Println(result)
}

File: src/022/022.rs
use std::fs::File;
use std::io::Read;

fn solve() -> i32 {
    let filename = "p022_names.txt";
    let mut file = File::open(filename).expect("File not found.");
    let mut contents = String::new();
    file.read_to_string(&mut contents).expect("Error reading file.");

    let filtered_name_list = contents.replace("\"", " ").replace(" ", "");
    let mut names = filtered_name_list.split(',').collect::<Vec<&str>>();
    names.sort();

    names
        .iter()
        .enumerate()
        .map(|(i, name)| {
            let bytes: i32 = name.bytes().map(|c| (c as i32) - 64).sum();
            let index = (i + 1) as i32;
            bytes * index
        })
        .sum()
}

fn main() {
    let result = solve();
    println!("{}", result);
}
